spec/compiler/crystal/tools/doc/project_info_spec.cr

```diff
@@ -33,7 +33,17 @@ describe Crystal::Doc::ProjectInfo do
         File.write("shard.yml", "name: foo\nversion: 1.0")
       end
 
-      it "no git" do
+      it "git missing" do
+        Crystal::Git.executable = "git-missing-executable"
+
+        assert_with_defaults(ProjectInfo.new(nil, nil), ProjectInfo.new("foo", "1.0", refname: nil))
+        assert_with_defaults(ProjectInfo.new("bar", "2.0"), ProjectInfo.new("bar", "2.0", refname: nil))
+        assert_with_defaults(ProjectInfo.new(nil, "2.0"), ProjectInfo.new("foo", "2.0", refname: nil))
+      ensure
+        Crystal::Git.executable = "git"
```

Comment:
I confirm that the specs get stuck immediately after exiting this spec. But I can't figure out where. Both the `around_each` and `before_each` above also finish, and *then* somehow spec gets stuck, before entering the next spec.

Reply:
We should probably just add a quick fix to disable this spec on win32 to get CI working.

Reply:
I think I understand now. Of course, the bug in entirely in Crystal's support of Windows.

This leaves a dangling fiber that's still trying to write to the non-launched process. While background fibers getting permanently stuck is a known (horrible) bug, perhaps there's another bug that forgets to close something if the process never actually launched.

Repro in `cmd`:

```cmd
crystal eval "p! Process.run(%(missing), output: IO::Memory.new) rescue nil; p! Process.run(%(git), output: IO::Memory.new)"
```

Reply:
@straight-shoota Yes, and looks like that'll even be the somewhat long-term fix. #9913